### PR TITLE
Use kekrby's audio config files

### PR DESCRIPTION
**Pull request by AdityaGarg8**

### CODE REVIEW

File: docs/guides/audio-config.md
Comment (Redecorating) [inline docs/guides/audio-config.md:32]: can this section be removed? Or replaced with just a note on pulseaudio?

**Comment by networkException** [comment]
Does this change anything about state.md?

**Comment by AdityaGarg8** [comment]
I couldn’t get automatic switching between headphones and speakers in state.

**Review by Redecorating — COMMENTED** [review]
I would like to figure out what needs to be changed to make my internal speakers get mapped correctly before we merge (e.g. atm "front right" is on the left, "front center" is on the right, and "Subwoofer" is static).

File: docs/guides/audio-config.md
Comment (AdityaGarg8) [reply docs/guides/audio-config.md:32]: We need this section cause Pulseaudio doesn't work with these?

**Comment by AdityaGarg8** [comment]
> I would like to figure out what needs to be changed to make my internal speakers get mapped correctly before we merge (e.g. atm "front right" is on the left, "front center" is on the right, and "Subwoofer" is static).

Thats strange, cause its mapped correctly for me.

File: docs/guides/audio-config.md
Comment (Redecorating) [reply docs/guides/audio-config.md:32]: Oh it's still not working with pulseaudio. Maybe include a link to the previous version of this page for those who don't want to switch?

**Comment by Redecorating** [comment]
> Thats strange, cause its mapped correctly for me.

Hmm I've been checking in the Gnome Settings Audio test thing, does it work properly for you there?
image

**Comment by AdityaGarg8** [comment]
> > Thats strange, cause its mapped correctly for me.
> 
> Hmm I've been checking in the Gnome Settings Audio test thing, does it work properly for you there? image

It did work iirc. I'll still check it again when I get time.

File: docs/guides/audio-config.md
Comment (AdityaGarg8) [reply docs/guides/audio-config.md:32]: Well, the previous version won't work as bce is also gonna get patched.

Comment (Redecorating) [reply docs/guides/audio-config.md:32]: Hmm they'd have to rename the files i guess. I'm not sure if we should break support for pa, would that be a regression?

Comment (AdityaGarg8) [reply docs/guides/audio-config.md:32]: We could use kekrby's old config files then.

Comment (kekrby) [reply docs/guides/audio-config.md:32]: [These files](https://github.com/kekrby/t2-better-audio/tree/163ecc6a0dbafbe8107fd5369bb374e343be9c61) should work with `pulseaudio` and an updated `apple-bce`.

Comment (kekrby) [reply docs/guides/audio-config.md:32]: A `pulseaudio` with [this patch](https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/merge_requests/596.patch) applied should work as well.

**Comment by kekrby** [comment]
I think how to check if `apple-bce` is updated should be included at the introduction part. `sed -n "s/.*\(AppleT2.*\) -.*/\1/p" /proc/asound/cards` can be used to check it as it will output `AppleT2xN` (where `N` is a number) if the driver has been updated and `AppleT2` if it hasn't. 

I can't suggest changes for that part as it was not edited in this PR so can you add this @AdityaGarg8?

**Review by kekrby — CHANGES_REQUESTED** [review]

File: docs/guides/audio-config.md
Comment (AdityaGarg8) [reply docs/guides/audio-config.md:23]: It was meant to :)

**Comment by kekrby** [comment]
> Does this change anything about state.md?

Audio works well now so I think it can be moved from "partially working" to "working".

**Comment by AdityaGarg8** [comment]
> > Does this change anything about state.md?
> 
> Audio works well now so I think it can be moved from "partially working" to "working".

Is the microphone volume fine for you?

**Comment by kekrby** [comment]
> Is the microphone volume fine for you?

The volume is pretty good for me.

**Comment by AdityaGarg8** [comment]
> > Is the microphone volume fine for you?
> 
> The volume is fine for me.

Still low for me :)

**Comment by kekrby** [comment]
> Still low for me :)
I think you can adjust it in settings, if that is sufficient by itself I don't think it is a big problem. If it requires an external program I guess it should still stay in "partially working" in state.md.

The part about switching to headphones not working should still be removed though.

**Comment by AdityaGarg8** [comment]
> > Still low for me :)
> 
> I think you can adjust it in settings, if that is sufficient by itself I don't think it is a big problem. If it requires an external program I guess it should still stay in "partially working" in state.md.
> 
> The part about switching to headphones not working should still be removed though.

Its written speakers and microphone (though I myself never tried the microphone)

**Comment by kekrby** [comment]
Then only the part about headphone switching should be removed from "not working".

File: docs/guides/audio-config.md
Comment (kekrby) [inline docs/guides/audio-config.md:34]:
```suggestion
# Using PulseAudio instead of PipeWire
```

I think it should be this way now since it tells how to use `PulseAudio` instead.

File: docs/guides/audio-config.md
Comment (kekrby) [inline docs/guides/audio-config.md:23]: I think this it makes more sense to put this in the introduction as the reader is told to check `apple-bce` there.

File: docs/guides/audio-config.md
Comment (kekrby) [inline docs/guides/audio-config.md:38]: I think it should be mentioned earlier that currently only `pipewire` works (without patching) and this should be put after that.

**Review by kekrby — CHANGES_REQUESTED** [review]

File: docs/guides/audio-config.md
Comment (AdityaGarg8) [reply docs/guides/audio-config.md:23]: Could you PR the changes you think?

**Comment by AdityaGarg8** [comment]
> > Thats strange, cause its mapped correctly for me.
> 
> Hmm I've been checking in the Gnome Settings Audio test thing, does it work properly for you there? image

@Redecorating , the mapping indeed is wrong for me as well.

File: docs/guides/audio-config.md
Comment (kekrby) [reply docs/guides/audio-config.md:23]: I opened #335 for the changes

**Comment by AdityaGarg8** [comment]
So any other changes required?

**Review by Redecorating — CHANGES_REQUESTED** [review]
I don't want to merge this until we either figure out how to fix the incorrect speaker mapping this introduces for 6 speaker macbooks, or we provide instructions on how to use the previous config files for 6 speaker makbooks so people won't have a regression in terms of the audio now being mapped wrong.

Other than fixing or providing advice to workaround that mapping issue for the 16 inch macbooks, it looks good.

**Review by kekrby — COMMENTED** [review]
I think this shouldn't get merged until the channel map for 6 speaker models is fixed and the `apple-bce` changes are merged to `t2linux/apple-bce-drv`.

**Comment by AdityaGarg8** [comment]
Let's wait for a fix then

**Comment by kekrby** [comment]
I think we can proceed with this now that the channel map issue is solved.

**Review by kekrby — CHANGES_REQUESTED** [review]
I think one last thing that should be done is making the capitalization of letters in PipeWire and PulseAudio consistent. Currently it is a mix of Pulseaudio/Pipewire and PulseAudio/PipeWire.

File: docs/guides/audio-config.md
Comment (AdityaGarg8) [reply docs/guides/audio-config.md:34]: I am having issues with pulse audio

**Comment by AdityaGarg8** [comment]
@Redecorating , could you merge it, if it looks good to you?

**Comment by AdityaGarg8** [comment]
BTW, should there be instructions to remove the old files as well? Or probably write some code to remove the old files in your script @kekrby ?

File: docs/guides/audio-config.md
Comment (kekrby) [inline docs/guides/audio-config.md:33]: These commands should delete the old files. 

Could you please wrap them in triple backticks? I didn't do it as it seemed to break the suggestion UI.

```suggestion

You should also run these commands to delete the old configuration files if you have already set up audio the old way:
sudo rm /usr/share/alsa/cards/AppleT2.conf
sudo rm /usr/lib/udev/rules.d/91-pulseaudio-custom.rules
sudo rm /usr/share/pulseaudio/alsa-mixer/profile-sets/apple-t2.conf

If you use PipeWire, run this command instead of the last one:
sudo rm /usr/share/alsa-card-profile/mixer/profile-sets/apple-t2.conf

```

**Review by kekrby — CHANGES_REQUESTED** [review]

**Comment by kekrby** [comment]
By the way now that there are a lot of commits that this PR includes, I think it makes sense to squash the related ones.

In my opinion 101ccca469b0ebed2998b187ee3763f95dfcc7ef should stay untouched, 0522c1ed3ed6f4f041a2b44220d81a28f38c1642 and 627dadc5c44ba476fcab60b5625e9a6ff3e70ebd should be squashed into one commit and the rest should be squashed into another commit as well.

File: docs/guides/audio-config.md
Comment (AdityaGarg8) [reply docs/guides/audio-config.md:33]: https://github.com/kekrby/t2-better-audio/pull/2

Check this out

File: docs/guides/audio-config.md
Comment (kekrby) [inline docs/guides/audio-config.md:30]: There is already a shebang line so I don't think this is necessary.

```suggestion
./install.sh
```

**Review by kekrby — CHANGES_REQUESTED** [review]

**Comment by AdityaGarg8** [comment]
> By the way now that there are a lot of commits that this PR includes, I think it makes sense to squash the related ones.
> 
> In my opinion [101ccca](https://github.com/t2linux/wiki/commit/101ccca469b0ebed2998b187ee3763f95dfcc7ef) should stay untouched, [0522c1e](https://github.com/t2linux/wiki/commit/0522c1ed3ed6f4f041a2b44220d81a28f38c1642) and [627dadc](https://github.com/t2linux/wiki/commit/627dadc5c44ba476fcab60b5625e9a6ff3e70ebd) should be squashed into one commit and the rest should be squashed into another commit as well.

Well as a general trend, I guess we shall be squash and merging the PR.

File: docs/guides/audio-config.md
Comment (Redecorating) [inline docs/guides/audio-config.md:30]: does this need sudo?

File: docs/guides/audio-config.md
Comment (Redecorating) [inline docs/guides/audio-config.md:31]: it was cloned without sudo so why is sudo needed to remove it? also this assumes the user was in their home directory when they cloned the repo

**Review by Redecorating — CHANGES_REQUESTED** [review]

File: docs/guides/audio-config.md
Comment (AdityaGarg8) [reply docs/guides/audio-config.md:31]: There is a strange issue on Ubuntu that causes error stating something like removing read only files if removing something from $HOME directory.

Comment (Redecorating) [reply docs/guides/audio-config.md:31]: hmm maybe it'd be simpler to clone it into /tmp? or use rm -rf?

Comment (AdityaGarg8) [reply docs/guides/audio-config.md]: 411234d872de947f2792fa3d23e0d26811fdc8eb

Also, I was thinking to attach the links of the apt repo as an easy alternative method.

Comment (AdityaGarg8) [reply docs/guides/audio-config.md]: You probably could also add the link for your Pacman repo?

File: docs/guides/audio-config.md
Comment (Redecorating) [reply docs/guides/audio-config.md:30]: it doesn't https://github.com/kekrby/t2-better-audio/blob/2d835c6e3d4fb0406d2933638d380c8b7fb92700/install.sh#L8

**Review by Redecorating — APPROVED** [review]

File: docs/guides/audio-config.md
Comment (AdityaGarg8) [reply docs/guides/audio-config.md:30]: Wdym?

Comment (Redecorating) [reply docs/guides/audio-config.md:30]: I mistakenly thought/remembered the script would say "please run me with sudo" and exit 1, but it doesn't do that so it's not an issue

File: docs/guides/audio-config.md
Comment (kekrby) [reply docs/guides/audio-config.md]: I think it should be mentioned that this guide should be followed only if the distro being used does not package the files as installing them as a distro package would allow updates to happen just like any other package but manual installation would not.

Comment (AdityaGarg8) [reply docs/guides/audio-config.md]: I guess we can make this change in a separate PR since I've to work on the apt repo a bit more.

Comment (AdityaGarg8) [reply docs/guides/audio-config.md]: I'll merge this if it's good apart from this.

Any other suggestions @Redecorating @kekrby?

**Review by kekrby — APPROVED** [review]
LGTM

File: docs/guides/audio-config.md
Comment (alex-007) [inline docs/guides/audio-config.md:28]: I don't think you should be a root to write something to /tmp. Isn't it better to not use `sudo` here?

Comment (sharpenedblade) [reply docs/guides/audio-config.md:28]: For certain distros it is necessary.

**Pull request merged**
